packages/verified-fetch/src/utils/request-headers.ts

```diff
@@ -0,0 +1,86 @@
+import { type CatOptions } from '@helia/unixfs'
+import { CodeError } from '@libp2p/interface'
+import { type ExporterOptions } from 'ipfs-unixfs-exporter'
+
+export function getHeader (headers: HeadersInit | undefined, header: string): string | undefined {
+  if (headers == null) {
+    return undefined
+  }
+  if (headers instanceof Headers) {
+    return headers.get(header) ?? undefined
+  }
+  if (Array.isArray(headers)) {
+    const entry = headers.find(([key]) => key.toLowerCase() === header.toLowerCase())
+    return entry?.[1]
+  }
+  const key = Object.keys(headers).find(k => k.toLowerCase() === header.toLowerCase())
+  if (key == null) {
+    return undefined
+  }
+
+  return headers[key]
+}
+
+/**
+ * Type abstraction that will break the build if the supported range options change.
+ */
+export interface HeliaRangeOptions extends Pick<ExporterOptions | CatOptions, 'offset' | 'length'> {
+  suffixLength?: number
+}
+
+/**
+ * Converts a range request header into helia/unixfs supported range options
+ * Note that the gateway specification says we "MAY" support multiple ranges (https://specs.ipfs.tech/http-gateways/path-gateway/#range-request-header) but we don't
+ *
+ * Also note that @helia/unixfs and ipfs-unixfs-exporter expect length and offset to be numbers, the range header is a string, and the size of the resource is likely a bigint.
+ *
+ * SUPPORTED:
+ * Range: bytes=<range-start>-<range-end>
+ * Range: bytes=<range-start>-
+ * Range: bytes=-<suffix-length> // must pass size so we can calculate the offset
+ *
+ * NOT SUPPORTED:
+ * Range: bytes=<range-start>-<range-end>, <range-start>-<range-end>
+ * Range: bytes=<range-start>-<range-end>, <range-start>-<range-end>, <range-start>-<range-end>
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range#directives
+ */
+export function getRequestRange (headers: Headers | HeadersInit | undefined, size?: bigint): HeliaRangeOptions | undefined {
```

Comment:
Can we simplify the input types here and just make it `Headers`?

Reply:
I was trying to normalize the request Header processing and response header generation further with the existing code.

Unfortunately, the code I have now is already hard for *me* to adjust (and too many changes needed in too many places), and I wrote it, so I started creating a `byte-range-context.ts` file with a class `ByteRangeContext` that should be able to handle things more easily. When that's done, much of the byte-range handling should be much more straightforward.

Right now, calling getRequestRange, and getContentRangeHeader, and all the possible response values require too much context within handleDagPb & handleRaw, so I think this will get much better

Reply:
function is removed and ByteRangeContext now handles parsing of header info upon construction.